### PR TITLE
View attached boxes in product view

### PR DESCRIPTION
**Pull request by Jokauppi**
- List for boxes in product view
- Reordered product view
- Placeholder for popularity graph
- Added missing box queries
- Decreased padding and fixed some layouts throughout the app

### CODE REVIEW

File: src/app/admin/products/ProductTable.tsx [hunk 1]
```diff
@@ -33,30 +33,37 @@ function ProductTable({ products }: { products: Product[] }) {
       return true;
     });
 
+  const sortedProducts = filteredProducts.sort((a, b) => {
```
Comment (ConcernedHobbit) [inline src/app/admin/products/ProductTable.tsx:36]: Consider `String.localeCompare` or `Intl.Collator` comparison. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator

File: src/app/admin/products/[barcode]/AttachedBoxesList.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
+"use const";
```
Comment (ConcernedHobbit) [inline src/app/admin/products/[barcode]/AttachedBoxesList.tsx:1]: "use client"? :D

File: src/app/admin/products/[barcode]/AttachedBoxesList.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+import Barcode from "@/components/Barcode";
+import { Button } from "@/components/ui/button";
+import { Box } from "@/server/requests/boxRequests";
+import Link from "next/link";
+
+type OwnProps = {
```
Comment (ConcernedHobbit) [inline src/app/admin/products/[barcode]/AttachedBoxesList.tsx:8]: nit: "AttachedBoxesListProps" is a handful but makes it clear when peeking in editor

File: src/app/admin/products/[barcode]/AttachedBoxesList.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        <Link href={`/admin/new/box?barcode=${barcode}`}>Attach Box</Link>
+      </Button>
+      <div className="-mt-2 h-full min-h-0 overflow-y-auto overscroll-none ">
+        <div className="flex flex-col gap-y-4 pb-8 pt-6">
+          {boxes.length === 0 && (
+            <p className="text-center text-stone-500">No boxes attached</p>
+          )}
+          {boxes.map((box, idx) => (
+            <div
+              key={idx}
```
Comment (ConcernedHobbit) [inline src/app/admin/products/[barcode]/AttachedBoxesList.tsx:26]: Prefer `box.boxBarcode` for key, as id can change depending on array sort

File: src/app/admin/products/[barcode]/ProductView.tsx [hunk 5]
```diff
@@ -1,66 +1,85 @@
...
+              <Label label="Stock">
+                <p
+                  id="stock"
+                  className={product.stock < 0 ? "text-red-500" : ""}
+                >
+                  {product.stock}
+                </p>
+              </Label>
+              <Label label="Buy Price">
+                <p id="buyPrice">{(product.buyPrice / 100).toFixed(2)} €</p>
```
Comment (ConcernedHobbit) [inline src/app/admin/products/[barcode]/ProductView.tsx:34]: Consider `currencyFormatter` (using `Intl.NumberFormat` with currency stule)

File: src/app/admin/products/[barcode]/ProductView.tsx [hunk 6]
```diff
@@ -1,66 +1,85 @@
...
+              </Label>
+              <Label label="Sell Price">
+                <p id="sellPrice">{(product.sellPrice / 100).toFixed(2)} €</p>
```
Comment (ConcernedHobbit) [inline src/app/admin/products/[barcode]/ProductView.tsx:37]: Consider `currencyFormatter` (using `Intl.NumberFormat` with currency style)

File: src/components/WithLabel.tsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
+"use client";
+
+import { merge } from "@/lib/utils";
+
+const stringToCamelcase = (str: string) =>
```
Comment (ConcernedHobbit) [inline src/components/WithLabel.tsx:5]: Consider moving this to utils

File: src/app/admin/products/[barcode]/AttachedBoxesList.tsx (hunk 2 above)
Comment (Jokauppi) [reply src/app/admin/products/[barcode]/AttachedBoxesList.tsx:1]: "use :D"

File: src/app/admin/products/[barcode]/AttachedBoxesList.tsx (hunk 3 above)
Comment (Jokauppi) [reply src/app/admin/products/[barcode]/AttachedBoxesList.tsx:8]: Keeping as is

**Pull request merged**